dali/python/backend_impl.cc

```diff
@@ -1117,6 +1119,27 @@ PYBIND11_MODULE(backend_impl, m) {
 
   m.def("GetCxx11AbiFlag", &GetCxx11AbiFlag);
 
+  m.def("HasCudaContext", []{
+    CUcontext context;
+    DALI_ENFORCE(cuInitChecked(),
+    "Failed to load libcuda.so. "
+    "Check your library paths and if the driver is installed correctly.");
+    CUDA_CALL(cuCtxGetCurrent(&context));
+    return context != nullptr;
+  });
+
+  py::class_<SharedMem>(m, "SharedMem")
+      .def(py::init<int, int>())
+      .def_property_readonly("size", &SharedMem::size)
+      .def_property_readonly("fd", &SharedMem::fd)
+      .def("buf",
+        [](SharedMem *shm) {
+          auto ptr = shm->ptr();
```

Comment:
How do you know if shm != nullptr. I would add a check.

Reply:
added the check